embassy-executor/src/arch/std.rs

```diff
@@ -62,6 +63,23 @@ mod thread {
                 self.signaler.wait()
             }
         }
+
+        /// Run the executor until a flag is raised.
+        ///
+        /// This function is identical to `Executor::run()` apart from offering a `done` flag to stop execution.
+        pub fn run_until(&'static mut self, done: &'static AtomicBool, init: impl FnOnce(Spawner)) {
```

Comment:
it'd be simpler to do `done: impl FnMut() -> bool`. This way the user can use any synchronization method, not necessarily an AtomicBool.